### ISSUE TITLE
[ISSUE]: Stuck at 100% CPU when loading a model

### ISSUE THREAD

**Issue by VelorumS** [deiteris/voice-changer#220]
### Voice Changer Version

b9cd071ae2743c146e5d279453a1c0caeb78e854

### Operational System

Manjaro 6.12.41-1

### GPU

RTX 4070 Ti

### Read carefully and check the options

- [x] I've tried to Clear Settings
- [ ] Sample/Default Models are working
- [ ] I've tried to change the Chunk Size
- [x] GUI was successfully launched
- [x] I've read the [tutorial](https://github.com/w-okada/voice-changer/blob/master/tutorials/tutorial_rvc_en_latest.md)
- [ ] I've tried to extract to another folder (or re-extract) the .zip file

### Model Type

RVC

### Issue Description

I've switched my local copy of the repo to the master-custom branch b9cd071ae2743c146e5d279453a1c0caeb78e854 (earlier was on fd849db23964158789fcaf1de93142f8714f90ce), created new venv with requirements-common.txt and requirements-cuda.txt. Loading a model gets stuck, can end only with SIGKILL.

I've tried a sample model and models I've used before.

### Logs on console

<code>~/voice/repos/voice-changer/server$ python3 main.py -p 18888 --https true 
2025-08-17 18:23:19,748 INFO     [main] Python: 3.10.18 (main, Aug 17 2025, 16:51:26) [GCC 15.1.1 20250729]
2025-08-17 18:23:19,748 INFO     [main] Voice changer version: Development -
2025-08-17 18:23:19,748 INFO     [WeightDownloader] Loading weights.
2025-08-17 18:23:19,791 INFO     [Downloader] Verified pretrain/crepe_onnx_full.onnx
2025-08-17 18:23:19,792 INFO     [Downloader] Verified pretrain/crepe_onnx_tiny.onnx
2025-08-17 18:23:19,830 INFO     [Downloader] Verified pretrain/crepe_full.pth
2025-08-17 18:23:19,831 INFO     [Downloader] Verified pretrain/crepe_tiny.pth
2025-08-17 18:23:19,985 INFO     [Downloader] Verified pretrain/content_vec_500.onnx
2025-08-17 18:23:20,061 INFO     [Downloader] Verified pretrain/rmvpe.pt
2025-08-17 18:23:20,210 INFO     [Downloader] Verified pretrain/rmvpe.onnx
2025-08-17 18:23:20,229 INFO     [Downloader] Verified pretrain/fcpe.pt
2025-08-17 18:23:20,248 INFO     [Downloader] Verified pretrain/fcpe.onnx
2025-08-17 18:23:20,249 INFO     [WeightDownloader] All weights are loaded!
2025-08-17 18:23:20,379 INFO     [main] protocol: HTTPS(self-signed), key:keys/20250817_182320.key, cert:keys/20250817_182320.cert
2025-08-17 18:23:22,965 INFO     [loader] Loading faiss with AVX2 support.
2025-08-17 18:23:22,965 INFO     [loader] Could not load library with AVX2 support due to:
ModuleNotFoundError("No module named 'faiss.swigfaiss_avx2'")
2025-08-17 18:23:22,965 INFO     [loader] Loading faiss.
2025-08-17 18:23:22,978 INFO     [loader] Successfully loaded faiss.
2025-08-17 18:23:23,256 INFO     [VoiceChangerManager] Initializing...
2025-08-17 18:23:23,309 INFO     [DeviceManager] Initialized DeviceManager. Backend statuses:
2025-08-17 18:23:23,309 INFO     [DeviceManager] * DirectML: False, device count: 0
2025-08-17 18:23:23,323 INFO     [DeviceManager] * CUDA: True, device count: 1
2025-08-17 18:23:23,323 INFO     [DeviceManager] * MPS: False
2025-08-17 18:23:23,342 INFO     [DeviceManager] Switched to 0: NVIDIA GeForce RTX 4070 Ti (CUDA) (cuda:0). FP16 support: True
2025-08-17 18:23:23,342 INFO     [IORecorder] Removing old recording file tmp_dir/in.wav
2025-08-17 18:23:23,342 INFO     [IORecorder] Removing old recording file tmp_dir/out.wav
2025-08-17 18:23:23,342 INFO     [IORecorder] -------------------------- - - - tmp_dir/in.wav, tmp_dir/out.wav
2025-08-17 18:23:23,519 INFO     [VoiceChangerV2] Allocated SOLA buffer size: 4800
2025-08-17 18:23:23,520 INFO     [VoiceChangerManager] Initialized.
2025-08-17 18:23:23,520 INFO     [VoiceChangerManager] Loading RVC...
2025-08-17 18:23:23,520 INFO     [RVCr2] Allocated audio buffer size: 9792
2025-08-17 18:23:23,520 INFO     [RVCr2] Allocated convert buffer size: 18080
2025-08-17 18:23:23,520 INFO     [RVCr2] Allocated pitchf buffer size: 114
2025-08-17 18:23:23,520 INFO     [RVCr2] Initializing...
~/voice/repos/voice-changer/.venv/lib/python3.10/site-packages/torch/nn/utils/weight_norm.py:144: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
2025-08-17 18:23:24,025 INFO     [RVCInferencerv2] Compiling JIT model...
2025-08-17 18:23:25,541 INFO     [EmbedderManager] Loading embedder hubert_base
2025-08-17 18:23:26,085 INFO     [PitchExtractorManager] Loading pitch extractor rmvpe_onnx
2025-08-17 18:23:26,815 INFO     [PipelineGenerator] Loading index...
2025-08-17 18:23:26,815 INFO     [PipelineGenerator] Try loading "model_dir/0/added_IVF5181_Flat_nprobe_1_v2.index.bin"...
~/voice/repos/voice-changer/.venv/lib/python3.10/site-packages/faiss/contrib/torch_utils.py:51: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  x.storage().data_ptr() + x.storage_offset() * 4)
</code>

**Comment by VelorumS** [comment]
Is this the same as https://github.com/deiteris/voice-changer/issues/216 ? I don't remember anything like that when I used voice-changer before.

**Comment by onestardao** [comment]
quick take — this screams “gpu fallback → cpu hot-loop” rather than a pure app bug.

likely culprits (short):

* torch not built with CUDA that matches your driver → model loads on CPU
* bitsandbytes / 8bit libs missing or incompatible → heavy CPU fallback
* `CUDA_VISIBLE_DEVICES` or driver issue (nvidia-smi shows nothing)
* too many OpenMP threads (OMP\_NUM\_THREADS) causing 100% core saturation during mmap/quant operations

quick checks (do these and paste results):

```bash
nvidia-smi
python -c "import torch,sys; print(torch.__version__, torch.cuda.is_available(), torch.cuda.device_count())"
pip show bitsandbytes accelerate | sed -n '1,5p'
echo $CUDA_VISIBLE_DEVICES $OMP_NUM_THREADS
```

quick fixes to try now:

* ensure you installed a CUDA-enabled torch that matches your driver (don’t use cpu wheel).
* install bitsandbytes if the model expects 8bit (pip install bitsandbytes).
* set `OMP_NUM_THREADS=1` before running to avoid thread thrash.
* try loading a very small model first to confirm GPU path works.

if you want I can map this to a concise troubleshooting checklist — paste the three outputs above and I’ll point the exact command you need.

**Comment by VelorumS** [comment]
```
Fri Aug 22 09:25:00 2025
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 575.64.05              Driver Version: 575.64.05      CUDA Version: 12.9     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4070 Ti     Off |   00000000:07:00.0  On |                  N/A |
|  0%   44C    P3             29W /  285W |    1638MiB /  12282MiB |      1%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
$ python -c "import torch,sys; print(torch.__version__, torch.cuda.is_available(), torch.cuda.device_count())"
2.8.0+cu128 True 1
$ pip show bitsandbytes accelerate
WARNING: Package(s) not found: accelerate, bitsandbytes
$ echo $CUDA_VISIBLE_DEVICES $OMP_NUM_THREADS

```

When I tried to reset voice-changer's settings, it defaults to CPU, and there it loads fast. These models are tiny.

There is `requirements-cuda.txt`, I assume that it takes care of the "CUDA-enabled torch".

bitsandbytes and `OMP_NUM_THREADS=1` don't change anything.

I've got a 55MiB model.

I may try to reinstall what worked for me before (the parent w-okada/voice-changer repo), that's more complicated than the fork.

Anyone tried the voice-changer on NVIDIA recently?

**Comment by onestardao** [comment]
Hi, it's faster for me to use AI and my system WFGY to reply you , sorry for it 

---
here’s the short answer. your logs show `torch.cuda.is_available() = False` and `device_count() = 0`. `nvidia-smi` is fine (driver 555.85, CUDA 12.6), so the Python wheels you installed are CPU-only or mismatched with the driver. the app then falls back to CPU and sticks there.

try this minimal sanity probe first:

```bash
python - <<'PY'
import torch, onnxruntime as ort
print('torch', torch.__version__, 'cuda?', torch.cuda.is_available(),
      'devs', torch.cuda.device_count(), 'torch.cuda', torch.version.cuda)
print('ort', ort.__version__, 'providers', ort.get_available_providers())
PY
```

if `cuda?` is False or providers lacks `CUDAExecutionProvider`, reinstall **GPU wheels** that match your driver.

**option A, CUDA 12.1 wheels (most stable on Windows):**

```bash
pip uninstall -y torch torchvision torchaudio onnxruntime onnxruntime-gpu
pip install --index-url https://download.pytorch.org/whl/cu121 torch torchvision torchaudio
pip install onnxruntime-gpu==1.17.3
```

**option B, CUDA 11.8 wheels:**

```bash
pip uninstall -y torch torchvision torchaudio onnxruntime onnxruntime-gpu
pip install --index-url https://download.pytorch.org/whl/cu118 torch torchvision torchaudio
pip install onnxruntime-gpu==1.16.3
```

then re-run the probe. you should see `cuda? True`, `devs >= 1`, and `CUDAExecutionProvider` present.
also check:

* `CUDA_VISIBLE_DEVICES` is not empty.
* your config selects GPU, not CPU.
* try a tiny model first to rule out OOM.

once the probe is green the voice-changer should stop pinning the CPU.

**Comment by VelorumS** [comment]
Please, ban AI junk bots in comments.

**Comment by onestardao** [comment]
Hi , I am here to help, I have helped lots of ppl from my method. if you don't like it, then I will leave this questions. (even I use AI that is from my method's observation)

anyway good luck for you. I will leave it to maintainer now
